optimade/filtertransformers/base_transformer.py

```diff
@@ -82,9 +199,58 @@ def not_implemented_string(self, value):
         """
         raise NotImplementedError("Comparing strings is not yet implemented.")
 
-    def property(self, arg):
-        """property: IDENTIFIER ( "." IDENTIFIER )*"""
-        return ".".join(arg)
+    def property(self, args):
+        """property: IDENTIFIER ( "." IDENTIFIER )*
+
+        If this transformer has an associated mapper, the property
+        will be compared to possible relationship entry types and
+        for any supported provider prefixes. If there is a match,
+        this rule will return a string and not a dereferenced
+        [`Quantity`][optimade.filtertransformers.base_transformer.Quantity].
+
+        Raises:
+            BadRequest: If the property does not match any
+                of the above rules.
+
+        """
+        quantity_name = str(args[0])
+
+        # If the quantity name matches an entry type (indicating a relationship filter)
+        # then simply return the quantity name; the inherited property
+        # must then handle any futher the nested identifiers
+        if self.mapper:
+            if quantity_name in self.mapper.RELATIONSHIP_ENTRY_TYPES:
+                return quantity_name
+
+        if self.quantities and quantity_name not in self.quantities:
+            # If the quantity is provider-specific, but does not match this provider,
+            # then return the quantity name such that it can be trreated as unknown.
+            # If the prefix does not match another known provider, also emit a warning
+            # If the prefix does match a known prpovider, do not return a warning.
+            # Following [Handling unknown property names](https://github.com/Materials-Consortia/OPTIMADE/blob/master/optimade.rst#handling-unknown-property-names)
+            if self.mapper and quantity_name.startswith("_"):
```

Comment:
I also think some of this should move into the ResourceMapper.

Also very similar code is used in the collection to verify include/exclude required properties.